cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -564,9 +565,11 @@ func (s *Supervisor) onOpampConnectionSettings(_ context.Context, settings *prot
 func (s *Supervisor) waitForOpAMPConnection() error {
 	// wait for the OpAMP client to connect to the server or timeout
 	select {
-	case <-s.connectedToOpAMPServer:
+	case s.connectedToOpAMPServer <- struct{}{}:
```

Comment:
```suggestion
	// This channel is unbuffered, so this will block until we listen on the channel in `OnConnectFunc`.
	case s.connectedToOpAMPServer <- struct{}{}:
```
Maybe not strictly necessary, but this is a slightly unconventional way to use channels (from what I've seen), so I think it would be good to leave a comment explaining why this works.

Someone else can feel free to chime in here to say we don't need this if you think its unlikely for someone to be tripped up by this in the future.

Reply:
It definitely confused me on first look.

Reply:
I'll definitely add a better comment explaining what's going on.

Do you mean that my change is an uncommon use of a channel? or the general pattern to synchronize the two goroutines? (golang newbie here).

I changed the direction of the channel because that way, the goroutine that creates it also writes to it and closes it (afaiu the preferred practice for managing channels). This also unblocks the connection task from waiting for the 'main' goroutine to synchronizing with it when the main goroutine is not waiting for a connection. Anyway I'll add a comment. Happy to make improvements if you think they'd help.